config/settings.rb

```diff
@@ -7,6 +7,79 @@
 SETTINGS.merge! YAML.load(ERB.new(File.read("#{root}/#{settings_file}")).result) if File.exist?(settings_file)
 SETTINGS[:version] = Foreman::Version.new
 
+# Load settings from env variables
+ENV_VARIABLE_SETTINGS_KEYS = {
+  'FOREMAN_UNATTENDED' => [:boolean, :unattended],
+  'FOREMAN_REQUIRE_SSL' => [:boolean, :require_ssl],
+  'FOREMAN_SUPPORT_JSONP' => [:boolean, :support_jsonp],
```

Comment:
Does this one really make sense? Can't we just check if jsonp can be loaded and automatically determine this?

I'd think that the following would raise an exception if it can't be loaded:

https://github.com/theforeman/foreman/blob/c5fbd28993fbe29f3457a329245e8975dd4c7667/config/application.rb#L68-L74

Otherwise we can check if `Rack::JSONP` is defined here:

https://github.com/theforeman/foreman/blob/c5fbd28993fbe29f3457a329245e8975dd4c7667/config/application.rb#L154-L155

Doing this would actually allow us to get rid of the setting altogether.

Reply:
I think we can get rid of this setting, but that's unrelated to this patch. If we want to remove it, let's do it in a separate PR.

Reply:
While I agree it should be a separate effort, I also wonder if including it in this list implies support. I'd consider not exposing the env var. The installer doesn't expose the setting and it defaults to false. That makes me wonder who uses this. Feels to me that jsonp is a technique that's no longer relevant and we could deprecate it.

Reply:
AFAIR this was introduced by ovirt project, maybe @masayag knows if its still being used. regardless its outside the scope of this PR (to remove it) and its pretty safe to not include it (As an env variable) if we don't want to encourage people using it.

Reply:
@ohadlevy I'm not familiar of any dependency of ovirt_provision_plugin on that set of env-variables.

Reply:
@masayag sorry I wasn't clear enough, the question here is if ovirt uses the JSONP feature of foreman.

Reply:
@ohadlevy yes, the entire interaction (requests & responses) from ovirt's side is done via JSON, i.e. `/api/v2/hosts?format=json&per_page=9999` is notation that states the JSONP feature is used.

https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/host/provider/foreman/ForemanHostProviderProxy.java#L37

Reply:
@masayag json and jsonp are not the same... see https://www.w3schools.com/js/js_json_jsonp.asp

Reply:
perhaps @ofavre remembers? :)